### PR TITLE
DXCDT-472: Add new `auth0_pages` resource

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

The `custom_login_page` on the `auth0_global_client` and the `change_password`, `guardian_mfa_page` and `error_page`
fields on the `auth0_tenant` have been deprecated in favour of managing them on a brand new `auth0_pages` resource.

This was done to improve the developer experience and discoverability of these configuration settings.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/auth0/page/resource.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Description: "Configuration settings for customizing the Password Reset page.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Indicates whether to use the custom Reset Password HTML (`true`) or the default Auth0 page (`false`). Defaults to `false`.",
```
Comment (willvedd) [inline internal/auth0/page/resource.go:65]:
```suggestion
							Description: "Indicates whether to use the custom Reset Password HTML (`true`) or the default Auth0 page (`false`).",
```

File: internal/auth0/page/resource.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Description: "Configuration settings for customizing the Login page.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Indicates whether to use the custom Login page HTML (`true`) or the default Auth0 page (`false`). Defaults to `false`.",
```
Comment (willvedd) [inline internal/auth0/page/resource.go:43]:
```suggestion
							Description: "Indicates whether to use the custom Login page HTML (`true`) or the default Auth0 page (`false`).",
```

File: internal/auth0/page/resource.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Description: "Configuration settings for customizing the Guardian Multi-Factor Authentication page.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Indicates whether to use the custom Guardian MFA HTML (`true`) or the default Auth0 page (`false`). Defaults to `false`.",
```
Comment (willvedd) [inline internal/auth0/page/resource.go:87]:
```suggestion
							Description: "Indicates whether to use the custom Guardian MFA HTML (`true`) or the default Auth0 page (`false`).",
```

File: internal/auth0/page/resource.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+		return diag.FromErr(err)
+	}
+
+	if err := api.Tenant.Update(&management.Tenant{
+		ChangePassword: &management.TenantChangePassword{
+			Enabled: auth0.Bool(false),
+		},
+		ErrorPage: &management.TenantErrorPage{
+			ShowLogLink: auth0.Bool(false),
+			URL:         auth0.String(""),
```
Comment (willvedd) [inline internal/auth0/page/resource.go:206]: Maybe clear out the HTML property too?

File: internal/auth0/page/resource_test.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+const testAccPagesCreate = `
```
Comment (willvedd) [inline internal/auth0/page/resource_test.go:11]: Perhaps consider adding a test case where we only modify a subset of the inner blocks?

File: internal/auth0/tenant/resource.go [hunk 6]
```diff
@@ -392,7 +407,7 @@ func NewResource() *schema.Resource {
 
 func createTenant(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	d.SetId(id.UniqueId())
-	return updateTenant(ctx, d, m)
+	return readTenant(ctx, d, m)
```
Comment (willvedd) [inline internal/auth0/tenant/resource.go:410]:
```suggestion
	return updateTenant(ctx, d, m)
```

File: MIGRATION_GUIDE.md [hunk 7]
```diff
@@ -1,5 +1,88 @@
 # Migration Guide
 
+## Upgrading from v0.49.0 → v0.50.0
+
+There are deprecations in this update. Please ensure you read this guide thoroughly and prepare your potential
+automated workflows before upgrading.
+
+### Deprecations
+
+- [Auth0 Pages](#auth0-pages)
```
Comment (willvedd) [inline MIGRATION_GUIDE.md:10]: IMO makes more sense to split out into two sections for tenant and global client.

File: internal/auth0/page/resource.go (hunk 4 above)
Comment (Widcket) [reply internal/auth0/page/resource.go:206]: Should this be done for the HTML properties of `login`, `change_password`, and `guardian_mfa` as well?

Comment (sergiught) [reply internal/auth0/page/resource.go:206]: As long as those pages are disabled, setting the HTML as well to empty is unnecessary. We preferred doing this as a user could in the future come back to these pages and re-enable them with the previous HTML.

**Review by Widcket — APPROVED** [review]

**Review by willvedd — APPROVED** [review]
👍

**Pull request merged**